proxmox/nodes/vms/custom_virtiofs_share.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package vms
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/bpg/terraform-provider-proxmox/proxmox/types"
+)
+
+// CustomVirtiofsShare handles Virtiofs directory shares.
+type CustomVirtiofsShare struct {
+	DirId       *string           `json:"dirid"                  url:"dirid"`
```

Comment:
I think this field is mandatory, so perhaps make it as a `string` value instead of pointer?